src/components/atoms/FloatButtonNav/index.tsx

```diff
@@ -1,31 +1,40 @@
 "use client";
 
 import { StaticImageData } from "next/image";
-import React from "react";
+import React, { useState } from "react";
 import Image from "next/image";
 import NextLink, { LinkProps } from "next/link";
 import { UrlObject } from "url";
 import { motion } from "framer-motion";
+import LoadingComponent from "../Loading";
 
 interface IFloatButtonNavProps {
   icon: StaticImageData;
   backgroundStyle: "main" | "secondary";
+  label?: string;
   href: UrlObject | string;
   hasCouponActive?: boolean;
 }
 
 const FloatButtonNav: React.FC<IFloatButtonNavProps & LinkProps<any>> = ({
   href,
+  label,
   backgroundStyle,
   icon,
   hasCouponActive = false,
 }) => {
+  const [stateLabel, setStateLabel] = useState<any>(label);
+
+  const onClick = () => {
+    setStateLabel(<LoadingComponent bgColor="none" fullSize={false} />);
```

Comment:
You u set a component in useState?
Has better ways to make this. 

Think with me: 

- When Loading component will be show?
- What's is default value at label? We need a state to use this?

Follow this thought: "if i need define a type with 'any' for my code not showing errors, i'm written a bad code."

If @renanpontez see this, he will talk this is a "gambiarra" rsrs''